### PR TITLE
Refactor timer in framework

### PR DESCRIPTION
**Pull request by sdww0**
1. Added comments to the PIT timer.
2. Move some functions in the APIC timer to the TSC module.
3. Set timer frequency to 1000 Hz

### CODE REVIEW

File: framework/aster-frame/src/arch/x86/kernel/tsc.rs [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
+/// The frequency of TSC(Hz)
+pub(crate) static TSC_FREQ: AtomicU64 = AtomicU64::new(0);
 
-/// Determine if the current system supports tsc_deadline mode.
-pub fn is_tsc_deadline_mode_supported() -> bool {
-    let cpuid = cpuid!(1);
-    (cpuid.ecx & TSC_DEADLINE_MODE_SUPPORT) > 0
+pub fn init_tsc_freq() {
+    let tsc_freq = determine_tsc_freq_via_cpuid()
+        .map_or(determine_tsc_freq_via_pit(), |freq| freq as u64 * 1000);
```
Comment (tatetian) [inline framework/aster-frame/src/arch/x86/kernel/tsc.rs:24]: So the two methods of `determine_tsc_freq_via_cpuid` and `determine_tsc_freq_via_pit` return frequencies in different units. Is it possible to eliminate this inconsistency? Manually converting the frequency unit here is error prone.

File: framework/aster-frame/src/arch/x86/timer/pit.rs [hunk 2]
```diff
@@ -1,21 +1,189 @@
...
+    /// Triggers an interrupt (only on channel 0) when the counter is terminated (1 -> 0).
+    /// The data port needs to be reset before the next interrupt.
+    /// ```text,ignore
+    ///            software reload counter
+    ///                      ⬇
+    ///               +------+             +----
+    ///               |      |             |
+    /// --------------+      +-------------+
+    /// ⬆            ⬆                   ⬆
+    /// init()   counter 1 -> 0         counter 1 -> 0
```
Comment (tatetian) [inline framework/aster-frame/src/arch/x86/timer/pit.rs:37]: Did you draw the diagram yourself? They look amazing.

File: framework/aster-frame/src/arch/x86/timer/mod.rs [hunk 3]
```diff
@@ -12,19 +12,20 @@ use spin::Once;
...
 
 use crate::arch::x86::kernel;
-use crate::config::TIMER_FREQ;
 use crate::sync::SpinLock;
 use crate::trap::IrqLine;
 
 use self::apic::APIC_TIMER_CALLBACK;
 
+/// The timer frequency (Hz)
+pub const TIMER_FREQ: u64 = 1000;
```
Comment (tatetian) [inline framework/aster-frame/src/arch/x86/timer/mod.rs:21]: So why is it preferable to choose this value? Justify the number in the comment.

File: framework/aster-frame/src/arch/x86/timer/pit.rs (hunk 2 above)
Comment (sdww0) [reply framework/aster-frame/src/arch/x86/timer/pit.rs:37]: Yes, I think that it is difficult to see the timer information directly with text description, so I have written these diagrams according to some other timer specifications I have seen before.

**Pull request merged**